ompi/mca/hook/comm_method/hook_comm_method_fns.c

```diff
@@ -567,12 +547,31 @@ ompi_report_comm_methods(int called_from_location) // 1 = from init, 2 = from fi
 // 1. the hoststring each host contributed
 // 2. the 2d table in method[] if it isn't too big
 // 3. summary of on/off host interconnect, and list the exceptions
-    if (myleaderrank == 0) {
+
 // 1: hoststring for each host
-        for (i=0; i<nleaderranks; ++i) {
-          printf("%s\n", allhoststrings[i]);
-        }
+    host_leader_printstring(hoststring, myleaderrank, nleaderranks, leader_comm, -1, -1);
+    if (myleaderrank == 0) {
         printf("\n");
+    }
+
+// 1b: affstring for each host
+    if (mca_hook_comm_method_enable_show_aff) {
+        if (myleaderrank == 0) {
+            printf("Affinity per host: (with ompi_display_comm_aff_columns %d)\n",
+                mca_hook_comm_method_aff_columns);
+        }
+        host_leader_printstring(affstring, myleaderrank, nleaderranks, leader_comm,
```

Comment:
affstring is only alloc'd from what I see when  `if (mylocalrank == 0) {` is true. is this the case here, or should its useage be checked here as well?

Reply:
Should be okay I think because all the non-leaders early return, so that's why those checks disappear after a certain point